File: models/Gpt-3.5-turbo/py/code/dynamic_programming/Elevator_Rides.py

n, x = map(int, input().split())
weights = list(map(int, input().split()))

weights.sort()  # Sort the weights in non-decreasing order

rides = 0
left = 0  # Index of the heaviest person on the left side of the elevator
right = n - 1  # Index of the lightest person on the right side of the elevator

while left <= right:
    if weights[left] + weights[right] <= x:
        # If the weights of the heaviest and lightest person can fit in one ride
        left += 1  # Move the index of the heaviest person towards the right
    right -= 1  # Move the index of the lightest person towards the left
    rides += 1  # Increment the number of rides

print(rides)

File: models/Gpt-3.5-turbo/cpp/code/problems/dynamic_programming/Elevator_Rides.cpp

#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main() {
    int n, x;
    cin >> n >> x;

    vector<int> weights(n);
    for (int i = 0; i < n; i++) {
        cin >> weights[i];
    }

    sort(weights.begin(), weights.end());

    int rides = 0;
    int current_weight = 0;
    int i = 0;

    while (i < n) {
        if (current_weight + weights[i] <= x) {
            current_weight += weights[i];
            i++;
        } else {
            rides++;
            current_weight = 0;
        }
    }

    if (current_weight > 0) {
        rides++;
    }

    cout << rides << endl;

    return 0;
}
